### DISCUSSION TITLE
`get-text --extent last_cmd_output` on same window

### DISCUSSION THREAD

**Discussion by karlb** [kovidgoyal/kitty#4962] [Q&A] [answered]
When executing `kitty @ get-text --extent last_cmd_output` as a command just by itself, the output is always a single empty line because the last command at that point in time is the `kitty @ get-text` command which did not return any output, yet.

This makes it hard to pipe the last command's output somewhere without executing the command again. Is there a way to do something like `kitty @ get-text --extent last_cmd_output | grep foo`, so that the output of the previous command will be grepped?

**Comment by page-down** [comment, marked as answer]
Do not try to get the command output while changing the last one.
The best way to do this is to open a new kitty window, use `--match` to match the previous window, and then get the last command output.

If you find it inconvenient and still want to execute the script in the current window, then you need to use the escape sequence to remove the last command output produced after the script execution, so that the previous command output becomes the last one, get the command output and then decide whether to mark the current script output as command output according to your needs.

I noticed this inconvenience immediately when using this feature, but had no motivation to add a new `previous_cmd_output` or make it accept parameters.

**Reply by karlb** [reply]
Thanks for the explanation!

> I noticed this inconvenience immediately when using this feature, but had no motivation to add a new previous_cmd_output or make it accept parameters.

Are PRs welcome for such a feature, or is it considered bloat/out of scope? Maybe having an extent for "last complete command output" would be slightly better then `previous_cmd_output`.

Looking at https://stackoverflow.com/questions/24283097/reusing-output-from-last-command-in-bash, it seems I am not the only one interested in such a command.

**Reply by kovidgoyal** [reply]
You are welcome to send a PR                                                                                                                       
                                                                                                                                                   
But note that what is left in the terminal may not be the actual output                                                                            
from the command since the command can output escape codes that wont be                                                                            
preserved, and there is a decoding of the command output and re-encoding                                                                           
as UTF-8.  Also if the output overflows the scrollback buffer it will br                                                                           
incomplete. So this is not the most robust way of doing it. However, it                                                                            
can be useful.

**Reply by karlb** [reply]
This is now implemented in https://github.com/kovidgoyal/kitty/commit/05098559307470af95992dc71adcd3039768f3d9.

**Comment by kovidgoyal** [comment]
You are welcome to send a PR

But note that what is left in the terminal may not be the actual output
from the command since the command can output escape codes that wont be
preserved, and there is a decoding of the command output and re-encoding
as UTF-8.  Also if the output overflows the scrollback buffer it will br
incomplete. So this is not the most robust way of doing it. However, it
can be useful.

**Reply by page-down** [reply]
> You are welcome to send a PR
> ...

For reference, the corresponding PR (closed and reimplemented)
https://github.com/kovidgoyal/kitty/pull/4973

Also, this exact same reply was already sent in April.
https://github.com/kovidgoyal/kitty/discussions/4962#discussioncomment-2566373

@kovidgoyal
Looks like you're cleaning out your email inbox. For some reason this email reply has been re-sent.

**Reply by kovidgoyal** [reply]
It isnt my inbox, it seems to be some kind of queue in github. At some point in the past email replies were not getting through, so I ended up replying using the webUI, looks like the email reply was finally delivered as well :)
